From 4dcab3c2904353c0a175765657316bbfc78af0d2 Mon Sep 17 00:00:00 2001
From: Nils Philippsen <nils@redhat.com>
Date: Thu, 25 Feb 2010 18:04:18 +0100
Subject: [PATCH] patch: gold

Squashed commit of the following:

commit afd331aca4c7cb4a0b53c7b0276253aab82424ae
Author: Nils Philippsen <nils@redhat.com>
Date:   Thu Feb 25 15:11:04 2010 +0100

    selection-to-path: explicitly specify library dependencies

commit db8abcd13a4d553d1f3e50e6fb0cc19a7c9f6ae5
Author: Nils Philippsen <nils@redhat.com>
Date:   Thu Feb 25 15:11:04 2010 +0100

    pagecurl: explicitly specify library dependencies

commit 9d4e2aad81e339ae3e2971b7c12d3ae12bd60220
Author: Nils Philippsen <nils@redhat.com>
Date:   Thu Feb 25 15:11:04 2010 +0100

    map-object: explicitly specify library dependencies

commit 257047325c0fb9297d764f34a6d4742fcfb837b7
Author: Nils Philippsen <nils@redhat.com>
Date:   Thu Feb 25 15:11:04 2010 +0100

    lighting: explicitly specify library dependencies

commit 709c245217a301652c1e5a3a288f5ae7c101aa33
Author: Nils Philippsen <nils@redhat.com>
Date:   Thu Feb 25 15:11:04 2010 +0100

    imagemap: explicitly specify library dependencies

commit a40311f72b994c5ffb37a206e7d40ffc099c8205
Author: Nils Philippsen <nils@redhat.com>
Date:   Thu Feb 25 15:11:04 2010 +0100

    ifs-compose: explicitly specify library dependencies

commit aebab1503e2065fe7665abd5818def52603e2eec
Author: Nils Philippsen <nils@redhat.com>
Date:   Thu Feb 25 15:11:04 2010 +0100

    gradient-flare: explicitly specify library dependencies

commit 0c7c425d1acdfecf64b453681df25cb08e344338
Author: Nils Philippsen <nils@redhat.com>
Date:   Thu Feb 25 15:11:04 2010 +0100

    gimpressionist: explicitly specify library dependencies

commit 2822fbfd9c0c1c099047d99d41ae9de8c8d52dc3
Author: Nils Philippsen <nils@redhat.com>
Date:   Thu Feb 25 15:11:04 2010 +0100

    gfig: explicitly specify library dependencies

commit 4c0fd7d3f63f25a112271422cecef1b8958a94f9
Author: Nils Philippsen <nils@redhat.com>
Date:   Thu Feb 25 15:11:04 2010 +0100

    fractal-explorer: explicitly specify library dependencies

commit 17e154a2856d0e97ebd76c706b6238fb3b40e74e
Author: Nils Philippsen <nils@redhat.com>
Date:   Thu Feb 25 15:36:20 2010 +0100

    flame: explicitly specify library dependencies

commit f7fadce58851854abb1ed54ce0141c8dc0434c12
Author: Nils Philippsen <nils@redhat.com>
Date:   Thu Feb 25 15:11:04 2010 +0100

    color-rotate: explicitly specify library dependencies

commit abbb6a2472e7e31e3aaa30c772f8f5dbaaf476c5
Author: Nils Philippsen <nils@redhat.com>
Date:   Wed Feb 24 15:40:50 2010 +0100

    script-fu: explicitly specify library dependencies

commit da5ee1fe718d9bbbda6c786ebf0f4ded43a7a246
Author: Manish Singh <yosh@gimp.org>
Date:   Wed Sep 9 17:41:20 2009 -0700

    Explicit shared lib deps to support gold

    (attempted cherry-picking from commit
    15497c1d6808b1ea479574b0b3e132484f7f172f, resolved conflicts)

    Signed-off-by: Nils Philippsen <nils@redhat.com>

commit 7cae0f249a5d2110a596f3fb22cdd33b07b05f1c
Author: Manish Singh <yosh@gimp.org>
Date:   Sun May 24 10:42:39 2009 -0700

    Explicitly specify library dependencies at link time, so we can use gold.
    (cherry picked from commit 582cb0f14eb9f145bd2a2f5c9fda12309ae0229f)

    Signed-off-by: Nils Philippsen <nils@redhat.com>
---
 libgimpthumb/Makefile.am               |    5 ++++-
 plug-ins/color-rotate/Makefile.am      |   12 +++++++-----
 plug-ins/common/Makefile.am            |   11 ++++++-----
 plug-ins/common/mkgen.pl               |   11 ++++++-----
 plug-ins/flame/Makefile.am             |   12 +++++++-----
 plug-ins/fractal-explorer/Makefile.am  |   12 +++++++-----
 plug-ins/gfig/Makefile.am              |   12 +++++++-----
 plug-ins/gimpressionist/Makefile.am    |   12 +++++++-----
 plug-ins/gradient-flare/Makefile.am    |   12 +++++++-----
 plug-ins/help-browser/Makefile.am      |    1 +
 plug-ins/ifs-compose/Makefile.am       |   12 +++++++-----
 plug-ins/imagemap/Makefile.am          |   12 +++++++-----
 plug-ins/lighting/Makefile.am          |   12 +++++++-----
 plug-ins/map-object/Makefile.am        |   12 +++++++-----
 plug-ins/metadata/Makefile.am          |    5 ++++-
 plug-ins/pagecurl/Makefile.am          |   12 +++++++-----
 plug-ins/script-fu/Makefile.am         |   14 ++++++++------
 plug-ins/selection-to-path/Makefile.am |   12 +++++++-----
 18 files changed, 113 insertions(+), 78 deletions(-)

diff --git a/libgimpthumb/Makefile.am b/libgimpthumb/Makefile.am
index 2e7b531..ee26219 100644
--- a/libgimpthumb/Makefile.am
+++ b/libgimpthumb/Makefile.am
@@ -86,7 +86,10 @@ noinst_PROGRAMS = gimp-thumbnail-list
 
 gimp_thumbnail_list_SOURCES = gimp-thumbnail-list.c
 
-gimp_thumbnail_list_LDADD = libgimpthumb-$(GIMP_API_VERSION).la
+gimp_thumbnail_list_LDADD = \
+	libgimpthumb-$(GIMP_API_VERSION).la \
+	$(GDK_PIXBUF_LIBS) \
+	$(GLIB_LIBS)
 
 
 install-data-local: install-ms-lib install-libtool-import-lib
diff --git a/plug-ins/color-rotate/Makefile.am b/plug-ins/color-rotate/Makefile.am
index c713c8c..f7e11b7 100644
--- a/plug-ins/color-rotate/Makefile.am
+++ b/plug-ins/color-rotate/Makefile.am
@@ -1,16 +1,18 @@
 ## Process this file with automake to produce Makefile.in
 
+if OS_WIN32
+mwindows = -mwindows
+else
+libm = -lm
+endif
+
 libgimpui = $(top_builddir)/libgimp/libgimpui-$(GIMP_API_VERSION).la
 libgimpconfig = $(top_builddir)/libgimpconfig/libgimpconfig-$(GIMP_API_VERSION).la
 libgimpwidgets = $(top_builddir)/libgimpwidgets/libgimpwidgets-$(GIMP_API_VERSION).la
 libgimp = $(top_builddir)/libgimp/libgimp-$(GIMP_API_VERSION).la
 libgimpcolor = $(top_builddir)/libgimpcolor/libgimpcolor-$(GIMP_API_VERSION).la
 libgimpbase = $(top_builddir)/libgimpbase/libgimpbase-$(GIMP_API_VERSION).la
-libgimpmath = $(top_builddir)/libgimpmath/libgimpmath-$(GIMP_API_VERSION).la
-
-if OS_WIN32
-mwindows = -mwindows
-endif
+libgimpmath = $(top_builddir)/libgimpmath/libgimpmath-$(GIMP_API_VERSION).la $(libm)
 
 AM_LDFLAGS = $(mwindows)
 
diff --git a/plug-ins/common/Makefile.am b/plug-ins/common/Makefile.am
index 66dc6dd..d58ecee 100644
--- a/plug-ins/common/Makefile.am
+++ b/plug-ins/common/Makefile.am
@@ -7,21 +7,22 @@
 ## Modify those two files instead of this one; for most
 ## plug-ins you should only need to modify plugin-defs.pl.
 
+if OS_WIN32
+mwindows = -mwindows
+else
+libm = -lm
+endif
 
 libgimp = $(top_builddir)/libgimp/libgimp-$(GIMP_API_VERSION).la
 libgimpbase = $(top_builddir)/libgimpbase/libgimpbase-$(GIMP_API_VERSION).la
 libgimpcolor = $(top_builddir)/libgimpcolor/libgimpcolor-$(GIMP_API_VERSION).la
 libgimpconfig = $(top_builddir)/libgimpconfig/libgimpconfig-$(GIMP_API_VERSION).la
-libgimpmath = $(top_builddir)/libgimpmath/libgimpmath-$(GIMP_API_VERSION).la
+libgimpmath = $(top_builddir)/libgimpmath/libgimpmath-$(GIMP_API_VERSION).la $(libm)
 libgimpmodule = $(top_builddir)/libgimpmodule/libgimpmodule-$(GIMP_API_VERSION).la
 libgimpui = $(top_builddir)/libgimp/libgimpui-$(GIMP_API_VERSION).la
 libgimpwidgets = $(top_builddir)/libgimpwidgets/libgimpwidgets-$(GIMP_API_VERSION).la
 
 
-if OS_WIN32
-mwindows = -mwindows
-endif
-
 AM_LDFLAGS = $(mwindows)
 
 libexecdir = $(gimpplugindir)/plug-ins
diff --git a/plug-ins/common/mkgen.pl b/plug-ins/common/mkgen.pl
index 40b4b74..b0cd786 100755
--- a/plug-ins/common/mkgen.pl
+++ b/plug-ins/common/mkgen.pl
@@ -51,21 +51,22 @@ print MK <<EOT;
 ## Modify those two files instead of this one; for most
 ## plug-ins you should only need to modify plugin-defs.pl.
 
+if OS_WIN32
+mwindows = -mwindows
+else
+libm = -lm
+endif
 
 libgimp = \$(top_builddir)/libgimp/libgimp-\$(GIMP_API_VERSION).la
 libgimpbase = \$(top_builddir)/libgimpbase/libgimpbase-\$(GIMP_API_VERSION).la
 libgimpcolor = \$(top_builddir)/libgimpcolor/libgimpcolor-\$(GIMP_API_VERSION).la
 libgimpconfig = \$(top_builddir)/libgimpconfig/libgimpconfig-\$(GIMP_API_VERSION).la
-libgimpmath = \$(top_builddir)/libgimpmath/libgimpmath-\$(GIMP_API_VERSION).la
+libgimpmath = \$(top_builddir)/libgimpmath/libgimpmath-\$(GIMP_API_VERSION).la \$(libm)
 libgimpmodule = \$(top_builddir)/libgimpmodule/libgimpmodule-\$(GIMP_API_VERSION).la
 libgimpui = \$(top_builddir)/libgimp/libgimpui-\$(GIMP_API_VERSION).la
 libgimpwidgets = \$(top_builddir)/libgimpwidgets/libgimpwidgets-\$(GIMP_API_VERSION).la
 
 
-if OS_WIN32
-mwindows = -mwindows
-endif
-
 AM_LDFLAGS = \$(mwindows)
 
 libexecdir = \$(gimpplugindir)/plug-ins
diff --git a/plug-ins/flame/Makefile.am b/plug-ins/flame/Makefile.am
index 262a9aa..417ea31 100644
--- a/plug-ins/flame/Makefile.am
+++ b/plug-ins/flame/Makefile.am
@@ -1,16 +1,18 @@
 ## Process this file with automake to produce Makefile.in
 
+if OS_WIN32
+mwindows = -mwindows
+else
+libm = -lm
+endif
+
 libgimpui = $(top_builddir)/libgimp/libgimpui-$(GIMP_API_VERSION).la
 libgimpconfig = $(top_builddir)/libgimpconfig/libgimpconfig-$(GIMP_API_VERSION).la
 libgimpwidgets = $(top_builddir)/libgimpwidgets/libgimpwidgets-$(GIMP_API_VERSION).la
 libgimp = $(top_builddir)/libgimp/libgimp-$(GIMP_API_VERSION).la
 libgimpcolor = $(top_builddir)/libgimpcolor/libgimpcolor-$(GIMP_API_VERSION).la
 libgimpbase = $(top_builddir)/libgimpbase/libgimpbase-$(GIMP_API_VERSION).la
-libgimpmath = $(top_builddir)/libgimpmath/libgimpmath-$(GIMP_API_VERSION).la
-
-if OS_WIN32
-mwindows = -mwindows
-endif
+libgimpmath = $(top_builddir)/libgimpmath/libgimpmath-$(GIMP_API_VERSION).la $(libm)
 
 AM_LDFLAGS = $(mwindows)
 
diff --git a/plug-ins/fractal-explorer/Makefile.am b/plug-ins/fractal-explorer/Makefile.am
index 16ae83a..b7b09af 100644
--- a/plug-ins/fractal-explorer/Makefile.am
+++ b/plug-ins/fractal-explorer/Makefile.am
@@ -1,16 +1,18 @@
 ## Process this file with automake to produce Makefile.in
 
+if OS_WIN32
+mwindows = -mwindows
+else
+libm = -lm
+endif
+
 libgimpui = $(top_builddir)/libgimp/libgimpui-$(GIMP_API_VERSION).la
 libgimpconfig = $(top_builddir)/libgimpconfig/libgimpconfig-$(GIMP_API_VERSION).la
 libgimpwidgets = $(top_builddir)/libgimpwidgets/libgimpwidgets-$(GIMP_API_VERSION).la
 libgimp = $(top_builddir)/libgimp/libgimp-$(GIMP_API_VERSION).la
 libgimpcolor = $(top_builddir)/libgimpcolor/libgimpcolor-$(GIMP_API_VERSION).la
 libgimpbase = $(top_builddir)/libgimpbase/libgimpbase-$(GIMP_API_VERSION).la
-libgimpmath = $(top_builddir)/libgimpmath/libgimpmath-$(GIMP_API_VERSION).la
-
-if OS_WIN32
-mwindows = -mwindows
-endif
+libgimpmath = $(top_builddir)/libgimpmath/libgimpmath-$(GIMP_API_VERSION).la $(libm)
 
 AM_LDFLAGS = $(mwindows)
 
diff --git a/plug-ins/gfig/Makefile.am b/plug-ins/gfig/Makefile.am
index b0a3ae7..8c25d59 100644
--- a/plug-ins/gfig/Makefile.am
+++ b/plug-ins/gfig/Makefile.am
@@ -1,16 +1,18 @@
 ## Process this file with automake to produce Makefile.in
 
+if OS_WIN32
+mwindows = -mwindows
+else
+libm = -lm
+endif
+
 libgimpui = $(top_builddir)/libgimp/libgimpui-$(GIMP_API_VERSION).la
 libgimpconfig = $(top_builddir)/libgimpconfig/libgimpconfig-$(GIMP_API_VERSION).la
 libgimpwidgets = $(top_builddir)/libgimpwidgets/libgimpwidgets-$(GIMP_API_VERSION).la
 libgimp = $(top_builddir)/libgimp/libgimp-$(GIMP_API_VERSION).la
 libgimpcolor = $(top_builddir)/libgimpcolor/libgimpcolor-$(GIMP_API_VERSION).la
 libgimpbase = $(top_builddir)/libgimpbase/libgimpbase-$(GIMP_API_VERSION).la
-libgimpmath = $(top_builddir)/libgimpmath/libgimpmath-$(GIMP_API_VERSION).la
-
-if OS_WIN32
-mwindows = -mwindows
-endif
+libgimpmath = $(top_builddir)/libgimpmath/libgimpmath-$(GIMP_API_VERSION).la $(libm)
 
 AM_LDFLAGS = $(mwindows)
 
diff --git a/plug-ins/gimpressionist/Makefile.am b/plug-ins/gimpressionist/Makefile.am
index bdca573..69ef428 100644
--- a/plug-ins/gimpressionist/Makefile.am
+++ b/plug-ins/gimpressionist/Makefile.am
@@ -1,19 +1,21 @@
 ## Process this file with automake to produce Makefile.in
 
+if OS_WIN32
+mwindows = -mwindows
+else
+libm = -lm
+endif
+
 libgimpui = $(top_builddir)/libgimp/libgimpui-$(GIMP_API_VERSION).la
 libgimpconfig = $(top_builddir)/libgimpconfig/libgimpconfig-$(GIMP_API_VERSION).la
 libgimpwidgets = $(top_builddir)/libgimpwidgets/libgimpwidgets-$(GIMP_API_VERSION).la
 libgimp = $(top_builddir)/libgimp/libgimp-$(GIMP_API_VERSION).la
 libgimpcolor = $(top_builddir)/libgimpcolor/libgimpcolor-$(GIMP_API_VERSION).la
 libgimpbase = $(top_builddir)/libgimpbase/libgimpbase-$(GIMP_API_VERSION).la
-libgimpmath = $(top_builddir)/libgimpmath/libgimpmath-$(GIMP_API_VERSION).la
+libgimpmath = $(top_builddir)/libgimpmath/libgimpmath-$(GIMP_API_VERSION).la $(libm)
 
 SUBDIRS = Brushes Paper Presets
 
-if OS_WIN32
-mwindows = -mwindows
-endif
-
 AM_CPPFLAGS = \
 	-DDEFAULTPATH=\""~/$(gimpdir)/gimpressionist:$(gimpdatadir)/gimpressionist"\"
 
diff --git a/plug-ins/gradient-flare/Makefile.am b/plug-ins/gradient-flare/Makefile.am
index 8597665..ecb8d75 100644
--- a/plug-ins/gradient-flare/Makefile.am
+++ b/plug-ins/gradient-flare/Makefile.am
@@ -1,16 +1,18 @@
 ## Process this file with automake to produce Makefile.in
 
+if OS_WIN32
+mwindows = -mwindows
+else
+libm = -lm
+endif
+
 libgimpui = $(top_builddir)/libgimp/libgimpui-$(GIMP_API_VERSION).la
 libgimpconfig = $(top_builddir)/libgimpconfig/libgimpconfig-$(GIMP_API_VERSION).la
 libgimpwidgets = $(top_builddir)/libgimpwidgets/libgimpwidgets-$(GIMP_API_VERSION).la
 libgimp = $(top_builddir)/libgimp/libgimp-$(GIMP_API_VERSION).la
 libgimpcolor = $(top_builddir)/libgimpcolor/libgimpcolor-$(GIMP_API_VERSION).la
 libgimpbase = $(top_builddir)/libgimpbase/libgimpbase-$(GIMP_API_VERSION).la
-libgimpmath = $(top_builddir)/libgimpmath/libgimpmath-$(GIMP_API_VERSION).la
-
-if OS_WIN32
-mwindows = -mwindows
-endif
+libgimpmath = $(top_builddir)/libgimpmath/libgimpmath-$(GIMP_API_VERSION).la $(libm)
 
 AM_LDFLAGS = $(mwindows)
 
diff --git a/plug-ins/help-browser/Makefile.am b/plug-ins/help-browser/Makefile.am
index 14fbe02..7360433 100644
--- a/plug-ins/help-browser/Makefile.am
+++ b/plug-ins/help-browser/Makefile.am
@@ -36,6 +36,7 @@ LDADD = \
 	$(libgimpbase)		\
 	$(WEBKIT_LIBS)		\
 	$(GIO_LIBS)		\
+	$(GLIB_LIBS)		\
 	$(RT_LIBS)		\
 	$(INTLLIBS)
 
diff --git a/plug-ins/ifs-compose/Makefile.am b/plug-ins/ifs-compose/Makefile.am
index 6e2de02..c758863 100644
--- a/plug-ins/ifs-compose/Makefile.am
+++ b/plug-ins/ifs-compose/Makefile.am
@@ -1,16 +1,18 @@
 ## Process this file with automake to produce Makefile.in
 
+if OS_WIN32
+mwindows = -mwindows
+else
+libm = -lm
+endif
+
 libgimpui = $(top_builddir)/libgimp/libgimpui-$(GIMP_API_VERSION).la
 libgimpconfig = $(top_builddir)/libgimpconfig/libgimpconfig-$(GIMP_API_VERSION).la
 libgimpwidgets = $(top_builddir)/libgimpwidgets/libgimpwidgets-$(GIMP_API_VERSION).la
 libgimp = $(top_builddir)/libgimp/libgimp-$(GIMP_API_VERSION).la
 libgimpcolor = $(top_builddir)/libgimpcolor/libgimpcolor-$(GIMP_API_VERSION).la
 libgimpbase = $(top_builddir)/libgimpbase/libgimpbase-$(GIMP_API_VERSION).la
-libgimpmath = $(top_builddir)/libgimpmath/libgimpmath-$(GIMP_API_VERSION).la
-
-if OS_WIN32
-mwindows = -mwindows
-endif
+libgimpmath = $(top_builddir)/libgimpmath/libgimpmath-$(GIMP_API_VERSION).la $(libm)
 
 AM_LDFLAGS = $(mwindows)
 
diff --git a/plug-ins/imagemap/Makefile.am b/plug-ins/imagemap/Makefile.am
index 1c06f94..0324900 100644
--- a/plug-ins/imagemap/Makefile.am
+++ b/plug-ins/imagemap/Makefile.am
@@ -1,16 +1,18 @@
 ## Process this file with automake to produce Makefile.in
 
+if OS_WIN32
+mwindows = -mwindows
+else
+libm = -lm
+endif
+
 libgimpui = $(top_builddir)/libgimp/libgimpui-$(GIMP_API_VERSION).la
 libgimpconfig = $(top_builddir)/libgimpconfig/libgimpconfig-$(GIMP_API_VERSION).la
 libgimpwidgets = $(top_builddir)/libgimpwidgets/libgimpwidgets-$(GIMP_API_VERSION).la
 libgimp = $(top_builddir)/libgimp/libgimp-$(GIMP_API_VERSION).la
 libgimpcolor = $(top_builddir)/libgimpcolor/libgimpcolor-$(GIMP_API_VERSION).la
 libgimpbase = $(top_builddir)/libgimpbase/libgimpbase-$(GIMP_API_VERSION).la
-libgimpmath = $(top_builddir)/libgimpmath/libgimpmath-$(GIMP_API_VERSION).la
-
-if OS_WIN32
-mwindows = -mwindows
-endif
+libgimpmath = $(top_builddir)/libgimpmath/libgimpmath-$(GIMP_API_VERSION).la $(libm)
 
 AM_LDFLAGS = $(mwindows)
 
diff --git a/plug-ins/lighting/Makefile.am b/plug-ins/lighting/Makefile.am
index 1bf79fc..0a7b463 100644
--- a/plug-ins/lighting/Makefile.am
+++ b/plug-ins/lighting/Makefile.am
@@ -1,17 +1,19 @@
 ## Process this file with automake to produce Makefile.in
 
+if OS_WIN32
+mwindows = -mwindows
+else
+libm = -lm
+endif
+
 libgimpui = $(top_builddir)/libgimp/libgimpui-$(GIMP_API_VERSION).la
 libgimpconfig = $(top_builddir)/libgimpconfig/libgimpconfig-$(GIMP_API_VERSION).la
 libgimpwidgets = $(top_builddir)/libgimpwidgets/libgimpwidgets-$(GIMP_API_VERSION).la
 libgimp = $(top_builddir)/libgimp/libgimp-$(GIMP_API_VERSION).la
 libgimpcolor = $(top_builddir)/libgimpcolor/libgimpcolor-$(GIMP_API_VERSION).la
-libgimpmath = $(top_builddir)/libgimpmath/libgimpmath-$(GIMP_API_VERSION).la
+libgimpmath = $(top_builddir)/libgimpmath/libgimpmath-$(GIMP_API_VERSION).la $(libm)
 libgimpbase = $(top_builddir)/libgimpbase/libgimpbase-$(GIMP_API_VERSION).la
 
-if OS_WIN32
-mwindows = -mwindows
-endif
-
 AM_LDFLAGS = $(mwindows)
 
 SUBDIRS = images
diff --git a/plug-ins/map-object/Makefile.am b/plug-ins/map-object/Makefile.am
index 2f7b7bc..62be8e8 100644
--- a/plug-ins/map-object/Makefile.am
+++ b/plug-ins/map-object/Makefile.am
@@ -1,17 +1,19 @@
 ## Process this file with automake to produce Makefile.in
 
+if OS_WIN32
+mwindows = -mwindows
+else
+libm = -lm
+endif
+
 libgimpui = $(top_builddir)/libgimp/libgimpui-$(GIMP_API_VERSION).la
 libgimpconfig = $(top_builddir)/libgimpconfig/libgimpconfig-$(GIMP_API_VERSION).la
 libgimpwidgets = $(top_builddir)/libgimpwidgets/libgimpwidgets-$(GIMP_API_VERSION).la
 libgimp = $(top_builddir)/libgimp/libgimp-$(GIMP_API_VERSION).la
 libgimpcolor = $(top_builddir)/libgimpcolor/libgimpcolor-$(GIMP_API_VERSION).la
-libgimpmath = $(top_builddir)/libgimpmath/libgimpmath-$(GIMP_API_VERSION).la
+libgimpmath = $(top_builddir)/libgimpmath/libgimpmath-$(GIMP_API_VERSION).la $(libm)
 libgimpbase = $(top_builddir)/libgimpbase/libgimpbase-$(GIMP_API_VERSION).la
 
-if OS_WIN32
-mwindows = -mwindows
-endif
-
 AM_LDFLAGS = $(mwindows)
 
 libexecdir = $(gimpplugindir)/plug-ins
diff --git a/plug-ins/metadata/Makefile.am b/plug-ins/metadata/Makefile.am
index dd9bb3d..935b069 100644
--- a/plug-ins/metadata/Makefile.am
+++ b/plug-ins/metadata/Makefile.am
@@ -55,11 +55,14 @@ INCLUDES = \
 	-I$(includedir)
 
 LDADD = \
+	$(libgimpui)		\
+	$(libgimpwidgets)	\
 	$(libgimp)		\
+	$(libgimpmath)		\
 	$(libgimpconfig)	\
 	$(libgimpcolor)		\
 	$(libgimpbase)		\
-	$(libgimpmath)		\
+	$(EXIF_LIBS)		\
 	$(GTK_LIBS)		\
 	$(RT_LIBS)		\
 	$(INTLLIBS)
diff --git a/plug-ins/pagecurl/Makefile.am b/plug-ins/pagecurl/Makefile.am
index 2a6eafb..2eeb422 100644
--- a/plug-ins/pagecurl/Makefile.am
+++ b/plug-ins/pagecurl/Makefile.am
@@ -1,17 +1,19 @@
 ## Process this file with automake to produce Makefile.in
 
+if OS_WIN32
+mwindows = -mwindows
+else
+libm = -lm
+endif
+
 libgimpui = $(top_builddir)/libgimp/libgimpui-$(GIMP_API_VERSION).la
 libgimpconfig = $(top_builddir)/libgimpconfig/libgimpconfig-$(GIMP_API_VERSION).la
 libgimpwidgets = $(top_builddir)/libgimpwidgets/libgimpwidgets-$(GIMP_API_VERSION).la
 libgimp = $(top_builddir)/libgimp/libgimp-$(GIMP_API_VERSION).la
 libgimpcolor = $(top_builddir)/libgimpcolor/libgimpcolor-$(GIMP_API_VERSION).la
-libgimpmath = $(top_builddir)/libgimpmath/libgimpmath-$(GIMP_API_VERSION).la
+libgimpmath = $(top_builddir)/libgimpmath/libgimpmath-$(GIMP_API_VERSION).la $(libm)
 libgimpbase = $(top_builddir)/libgimpbase/libgimpbase-$(GIMP_API_VERSION).la
 
-if OS_WIN32
-mwindows = -mwindows
-endif
-
 AM_LDFLAGS = $(mwindows)
 
 libexecdir = $(gimpplugindir)/plug-ins
diff --git a/plug-ins/script-fu/Makefile.am b/plug-ins/script-fu/Makefile.am
index 3075b83..824d62d 100644
--- a/plug-ins/script-fu/Makefile.am
+++ b/plug-ins/script-fu/Makefile.am
@@ -1,21 +1,23 @@
 ## Process this file with automake to produce Makefile.in
 
+if OS_WIN32
+mwindows = -mwindows
+WINSOCK_LIBS = -lws2_32
+else
+libm = -lm
+endif
+
 libgimpui = $(top_builddir)/libgimp/libgimpui-$(GIMP_API_VERSION).la
 libgimpwidgets = $(top_builddir)/libgimpwidgets/libgimpwidgets-$(GIMP_API_VERSION).la
 libgimp = $(top_builddir)/libgimp/libgimp-$(GIMP_API_VERSION).la
 libgimpcolor = $(top_builddir)/libgimpcolor/libgimpcolor-$(GIMP_API_VERSION).la
 libgimpbase = $(top_builddir)/libgimpbase/libgimpbase-$(GIMP_API_VERSION).la
 libgimpconfig = $(top_builddir)/libgimpconfig/libgimpconfig-$(GIMP_API_VERSION).la
-libgimpmath = $(top_builddir)/libgimpmath/libgimpmath-$(GIMP_API_VERSION).la
+libgimpmath = $(top_builddir)/libgimpmath/libgimpmath-$(GIMP_API_VERSION).la $(libm)
 
 libtinyscheme=tinyscheme/libtinyscheme.a
 libftx=ftx/libftx.a
 
-if OS_WIN32
-mwindows = -mwindows
-WINSOCK_LIBS = -lws2_32
-endif
-
 AM_CFLAGS = \
 	-DSTANDALONE=0     \
 	-DUSE_INTERFACE=1  \
diff --git a/plug-ins/selection-to-path/Makefile.am b/plug-ins/selection-to-path/Makefile.am
index 7e3ac65..194b5dc 100644
--- a/plug-ins/selection-to-path/Makefile.am
+++ b/plug-ins/selection-to-path/Makefile.am
@@ -1,16 +1,18 @@
 ## Process this file with automake to produce Makefile.in
 
+if OS_WIN32
+mwindows = -mwindows
+else
+libm = -lm
+endif
+
 libgimpui = $(top_builddir)/libgimp/libgimpui-$(GIMP_API_VERSION).la
 libgimpconfig = $(top_builddir)/libgimpconfig/libgimpconfig-$(GIMP_API_VERSION).la
 libgimpwidgets = $(top_builddir)/libgimpwidgets/libgimpwidgets-$(GIMP_API_VERSION).la
 libgimp = $(top_builddir)/libgimp/libgimp-$(GIMP_API_VERSION).la
 libgimpcolor = $(top_builddir)/libgimpcolor/libgimpcolor-$(GIMP_API_VERSION).la
 libgimpbase = $(top_builddir)/libgimpbase/libgimpbase-$(GIMP_API_VERSION).la
-libgimpmath = $(top_builddir)/libgimpmath/libgimpmath-$(GIMP_API_VERSION).la
-
-if OS_WIN32
-mwindows = -mwindows
-endif
+libgimpmath = $(top_builddir)/libgimpmath/libgimpmath-$(GIMP_API_VERSION).la $(libm)
 
 AM_LDFLAGS = $(mwindows)
 
-- 
1.6.6.1

